doc/source/overview/saving.md

```diff
@@ -120,28 +121,47 @@ cd = ClassifierDrift(x_ref, model, backend='sklearn', p_val=.05, preds_type='pro
 In order for a detector to be saveable and loadable, any models contained within it (or referenced within a 
 [detector configuration file](config_files.md#specifying-artefacts)) must fall within the family of supported models:
 
-````{tab-set}
+`````{tab-set}
+
+````{tab-item} PyTorch
+PyTorch models are serialized by saving the
+[entire model](https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-entire-model)
+using the [dill](https://dill.readthedocs.io/en/latest/index.html) module. Therefore, Alibi Detect should support 
+any PyTorch model that can be saved and loaded with `torch.save(..., pickle_module=dill)` and 
+`torch.load(..., pickle_module=dill)`.
+````
 
-```{tab-item} TensorFlow
-Alibi Detect supports serialization of any TensorFlow model that can be serialized to the 
-[HDF5](https://www.tensorflow.org/guide/keras/save_and_serialize#keras_h5_format) format. 
-Custom objects should be pre-registered with 
-[register_keras_serializable](https://www.tensorflow.org/api_docs/python/tf/keras/utils/register_keras_serializable).
+````{tab-item} TensorFlow
+TensorFlow models are serialized to the TensorFlow 
+[SavedModel](https://www.tensorflow.org/guide/saved_model) format. For loading, 
+{func}`~alibi_detect.saving.load_detector` supports both the TensorFlow 
+[SavedModel](https://www.tensorflow.org/guide/saved_model) and 
+[HDF5](https://www.tensorflow.org/guide/keras/save_and_serialize#keras_h5_format) formats.
+
+```{warning}
+If the model, or constituent layers, are constructed by 
+[subclassing](https://www.tensorflow.org/guide/keras/custom_layers_and_models):
+- the custom classes must be defined in the loading runtime and decorated with
+[register_keras_serializable](https://www.tensorflow.org/api_docs/python/tf/keras/utils/register_keras_serializable),
+or stored in a [custom_objects](https://www.tensorflow.org/guide/keras/save_and_serialize#custom_objects) dictionary 
+given to {func}`~alibi_detect.saving.load_detector`.
+- Custom objects must have valid `get_config` and `from_config` (see
+[here](https://www.tensorflow.org/tutorials/keras/save_and_load#saving_custom_objects)).
 ```
 
-```{tab-item} PyTorch
-PyTorch models are serialized by saving the [entire model](https://pytorch.org/tutorials/beginner/saving_loading_models.html#save-load-entire-model)
-using the [dill](https://dill.readthedocs.io/en/latest/index.html) module. Therefore, Alibi Detect should support any PyTorch 
-model that can be saved and loaded with `torch.save(..., pickle_module=dill)` and `torch.load(..., pickle_module=dill)`.
+```{note}
+
+- The {obj}`~alibi_detect.cd.tensorflow.HiddenOutput` utility class is not currently compatible with subclassed models.
```

Comment:
`HiddenOutput` works by building a new `tf.keras.Model` from the original model's `input` and `layers` attributes.

https://github.com/SeldonIO/alibi-detect/blob/c0c5e644a1faaadc23d375f31e27a13110ce6bac/alibi_detect/cd/tensorflow/preprocess.py#L67-L85

This works fine for `tf.keras.Model`'s constructed from `tf.keras.Sequential` etc as they have a pre-defined structure to their these attributes. It doesn't work out-of-the-box for subclassed models, presumably because there are quite a few different ways you can construct these models internally. Probably worth opening an issue to explore this one further...

Reply:
I see, feels like a docstring is required for `HiddenOutput`(and other utility classes) to describe behaivour/limitations. For another PR though.

Reply:
I'll open an issue shortly 👍🏻